bpf/include/bpf/config/node.h

```diff
@@ -15,4 +15,6 @@
 /* Legacy node config rendered at agent runtime. */
 #include <node_config.h>
 
+NODE_CONFIG(__u32, ipv4_loopback, "IPv4 loopback address")
+#define IPV4_LOOPBACK CONFIG(ipv4_loopback)  /* Backwards compatibility */
```

Comment:
IPV4_LOOPBACK only has a handful of references outside of tests, so I'd drop the macro and use `CONFIG(ipv4_loopback)` directly. The reason for some of the existing backwards compat macros is to avoid pinging all maintainers, but in this case we should be good.

Edit: looks like using __be32 for ipv4 addresses is very common in the bpf codebase, which may not be worth changing at the moment. In that case, I'd define `IPV4_LOOPBACK` as `bpf_ntohs(loopback_ipv4)` and make loopback_ipv4 a [4]byte.

Reply:
> IPV4_LOOPBACK only has a handful of references outside of tests, so I'd drop the macro and use CONFIG(ipv4_loopback) directly. The reason for some of the existing backwards compat macros is to avoid pinging all maintainers, but in this case we should be good.

Sure, I'll swap things out to use `CONFIG(ipv4_loopback)` explicitly (I wanted to make the diff small at first until someone reviewed -- I wasn't sure what the preference would be here).

> Edit: looks like using __be32 for ipv4 addresses is very common in the bpf codebase, which may not be worth changing at the moment. In that case, I'd define IPV4_LOOPBACK as bpf_ntohs(loopback_ipv4) and make loopback_ipv4 a [4]byte.

Will do.

Reply:
> Edit2: I'm confused by our use of byteorder.NetIPv4ToHost32 in the agent to pass the value down to the datapath. This should mean the __be32 saddr fields actually contain LE values? Or maybe the Go value formatter changes the order of the bytes? This is one of those issues I'd like to sidestep with this work altogether..

I believe that before, `WriteNodeConfig` used `NetIPv4ToHost32` when it wrote out the hex to `node_config.h` so that when the value was read back as a hex value it was correctly ordered in memory. I dropped its use altogether in the latest push, so a byte array should stay a byte array all the way through.